packages/mdc-menu-surface/README.md

```diff
@@ -0,0 +1,164 @@
+```
+
+## Usage
+
+### HTML Structure
+
+```html
+<div class="mdc-menu-surface">
+...
+</div>
+```
+
+### Styles
+
+```css
+@import "@material/menu-surface/mdc-menu-surface";
+```
+
+### JavaScript Instantiation
+
+```js
+import {MDCMenuSurface} from '@material/menu-surface';
+
+const menuSurface = new MDCMenuSurface(document.querySelector('.mdc-menu-surface'));
+```
+
+> See [Importing the JS component](../../docs/importing-js.md) for more information on how to import JavaScript.
+
+## Style Customization
+
+### CSS Classes
+
+CSS Class | Description
+--- | ---
+`mdc-menu-surface` | Mandatory.
+`mdc-menu-surface--animating-open` | Indicates the menu surface is currently animating open. This class is removed once the animation completes.
+`mdc-menu-surface--open` | Indicates the menu surface is currently open, or is currently animating open.
+`mdc-menu-surface--animating-closed` | Indicates the menu surface is currently animating closed. This class is removed once the animation completes.
+`mdc-menu-surface--anchor` | Used to indicate which element the menu should be anchored to.
+`mdc-menu-surface--fixed` | Used to indicate that the menu is using fixed positioning.
+
+### Sass Mixins
+
+Mixin | Description
+--- | ---
+`mdc-menu-surface-ink-color($color)` | Sets the `color` property of the `mdc-menu-surface`.
+`mdc-menu-surface-fill-color($color)` | Sets the `background-color` property of the `mdc-menu-surface`.
+
+## `MDCMenuSurface` Properties and Methods
+
+Property | Value Type | Description
+--- | --- | ---
+`open` | Boolean | Proxies to the foundation's `isOpen`/(`open`, `close`) methods.
+`quickOpen` | Boolean | Proxies to the foundation's `setQuickOpen()` method.
+
+Method Signature | Description
+--- | ---
+`show() => void` | Proxies to the foundation's `open()` method.
+`hide() => void` | Proxies to the foundation's `close()` method.
+`setAnchorCorner(Corner) => void` | Proxies to the foundation's `setAnchorCorner(Corner)` method.
+`setAnchorMargin(AnchorMargin) => void` | Proxies to the foundation's `setAnchorMargin(AnchorMargin)` method.
+`setFixedPosition(isFixed: boolean) => void` | Adds the `mdc-menu-surface--fixed` class to the `mdc-menu-surface` element. Proxies to the foundation's `setIsHoisted()` and `setFixedPosition()` methods.
+`setAbsolutePosition(x: number, y: number) => void` | Proxies to the foundation's `setAbsolutePosition(x, y)` method. Used to set the absolute x/y position of the menu on the page. Should only be used when the menu is hoisted to the body.
+`setMenuSurfaceAnchorElement(element: Element) => void` | Changes the element used as an anchor for `menu-surface` positioning logic. Should be used with conjunction with `hoistMenuToBody()`.
+`hoistMenuToBody() => void` | Removes the `menu-surface` element from the DOM and appends it to the `body` element. Should be used to overcome `overflow: hidden` issues.
+`setIsHoisted() => void` | Proxies tot he foundation's `setIsHoisted` method.
```

Comment:
This function sets the state of the foundation so calculate everything based on the body. Since the menu could technically be a few level deep into the body, but still be positioned relative to the body, the user may not always have the menu completely hoisted to the body.